cpp/src/parquet/encryption/encryption_internal.cc

```diff
@@ -758,9 +795,14 @@ void QuickUpdatePageAad(int32_t new_page_ordinal, std::string* AAD) {
   std::memcpy(AAD->data() + AAD->length() - 2, page_ordinal_bytes.data(), 2);
 }
 
-void RandBytes(unsigned char* buf, int num) {
+void RandBytes(unsigned char* buf, size_t num) {
+  if (num > static_cast<size_t>(std::numeric_limits<int>::max())) {
```

Comment:
Note we could easily call `RAND_bytes` in a loop, but we don't have a use case for generating cryptographic random data that large anyway.